doc/spec/Types.md

```diff
@@ -1322,6 +1352,81 @@ foo({ id: 1234, name: false });    // Error, name of wrong type
 foo({ name: "hello" });            // Error, id required but missing
 ```
 
+### Comparability
+
+*TODO: Document the base primitive type.*
+
+Types are required to be *comparable* in certain circumstances, such as part of when checking whether two values of given types might be equal at runtime using operators like '===', or when using a type assertion.
+
+*S* is ***comparable to*** a type *T*, and *T* is ***comparable from*** *S*, if *S* has no excess properties with respect to *T* ([#excess-properties]<!--3.11.5-->) and one of the following is true:
+
+* *S* and *T* are identical types.
+* *S* or *T* is the Any type.
+* *S* is the Never type.
+* *S* is the Undefined type and *T* is the Void type.
+* *T* is the Undefined type and *S* is the Void type.
+* *S* is either the Undefined or Null type, and strict null checks ([#strict-null-checks]) are disabled.
+* *S* is an object type and *T* is the non-primitive `object` type.
+* *T* is an object type and *S* is the non-primitive `object` type.
+* *S* or *T* is a numeric enum or numeric enum member type and the other is the primitive type Number.
+* *S* and *T* are both enum member types with respective containing enum types *E* and *F*, and
+    * *E* is comparable to *F*.
+    * *F* is comparable to *E*.
+* *S* and *T* are both non-const enum types with the same declared name, and
+  * for each enum member *M* in *E*, there exists an enum member declaration *N* in *F* such that *M* and *N* have the same declared name and value.
+  * for each enum member *N* in *F*, there exists an enum member declaration *M* in *E* such that *N* and *M* have the same declared name and value.
+* *S* is an enum member type, *T* is a literal type, and the literal values of *S* and *T* are identical.
+* *T* is an enum member type, *S* is a literal type, and the literal values of *S* and *T* are identical.
+* *S* is a literal type and *T* is the base primitive type of *S*.
+* *T* is a literal type and *S* is the base primitive type of *T*.
+* *S* is a union type and some constituent type of *S* is comparable to *T*.

```

Comment:
Is this relation 100% symmetric? Can we just say that up front and cut this in half?

Reply:
Or even if it's not, parcel out the symmetric relations, e.g. *S* is assignable to *T* if any of the following are true for `S* = S, T* = T` *or* for `S* = T, T* = S`... (all symmetric properties here)

Reply:
@RyanCavanaugh I don't quite understand what you mean. But I'd prefer not to combine these operations so that a reader (or a spec writer 😉) can manually do a diff between the assignability and comparability sections on their own